src/Microsoft.OData.Core/Metadata/MetadataUtils.cs

```diff
@@ -171,27 +171,15 @@ internal static IList<IEdmOperation> CalculateBindableOperationsForType(IEdmType
                 throw new ODataException(Strings.MetadataUtils_CalculateBindableOperationsForType(bindingType.FullTypeName()), exc);
             }
 
-            List<IEdmOperation> operationsFound = new List<IEdmOperation>();
-            foreach (IEdmOperation operation in operations)
-            {
-                if (!operation.IsBound)
-                {
-                    throw new ODataException(Strings.EdmLibraryExtensions_UnBoundOperationsFoundFromIEdmModelFindMethodIsInvalid(operation.Name));
-                }
+            IList<IEdmOperation> operationsFound = operations as IList<IEdmOperation>;
 
-                if (operation.Parameters.FirstOrDefault() == null)
-                {
-                    throw new ODataException(Strings.EdmLibraryExtensions_NoParameterBoundOperationsFoundFromIEdmModelFindMethodIsInvalid(operation.Name));
-                }
-
-                IEdmOperationParameter bindingParameter = operation.Parameters.FirstOrDefault();
-                IEdmType resolvedBindingType = edmTypeResolver.GetParameterType(bindingParameter).Definition;
-                if (resolvedBindingType.IsAssignableFrom(bindingType))
-                {
-                    operationsFound.Add(operation);
-                }
+            if(operationsFound != null)
+            {
+                return operationsFound;
             }
 
+            operationsFound = new List<IEdmOperation>(operations);

```

Comment:
This is the fall back option when the returned IEnumarable cannot be casted to IList. We are hoping this is a less common scenario

Reply:
With this fix, FindBoundOperations should both be less expensive, be cached, and be returned as an IList<T>, so (unless the developer overwrites FindBoundOperations) this code should never get hit.

---
In reply to: [493156324](https://github.com/OData/odata.net/pull/1888#discussion_r493156324) [](ancestors = 493156324)

Reply:
As above, the only time the returned IEnumerable will not be an IList<T> is if the developer overwrites FindBoundOperations(bindingType) and returns a non-IEnumerable IList<T>. This would be a very uncommon scenario, and the developer could address any perf issues  by returning an IList<T>.

---
In reply to: [493184421](https://github.com/OData/odata.net/pull/1888#discussion_r493184421) [](ancestors = 493184421)